### PR TITLE
Add TTL to snapshots.

### PR DESCRIPTION
**Pull request by wwwtyro**

### CODE REVIEW

**Comment by wwwtyro** [comment]
test this

File: scheduler/leader.go [hunk 1]
```diff
@@ -57,13 +62,81 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, dao dao.Cont
...
 	// starts all of the listeners
 	zzk.Start(shutdown, conn, serviceListener, hostRegistry, snapshotListener)
 }
 
+func cleanSnapshots(cpClient dao.ControlPlane) {
+
+	cleaningInterval := time.Tick(10 * time.Minute)
+
+	// Determine the SERVICED_SNAPSHOT_TTL environemnt variable and convert to duration.
+	TTLenv, err := strconv.ParseInt(os.Getenv("SERVICED_SNAPSHOT_TTL"), 10, 64)
```
Comment (smousa) [inline scheduler/leader.go]: see cli/api/daemon.go on how we handle os.GetEnv (so we can pass it in as env and cmdline arg)

File: scheduler/leader.go [hunk 2]
```diff
@@ -57,13 +62,81 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, dao dao.Cont
...
+	if err != nil {
+		TTLenv = 12
+	}
+	TTL := time.Duration(TTLenv) * time.Hour
```
Comment (smousa) [inline scheduler/leader.go]: this looks like a constant when the var is declared as all caps

File: scheduler/leader.go [hunk 3]
```diff
@@ -57,13 +62,81 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, dao dao.Cont
...
 	serviceListener := zkservice.NewServiceListener(&leader)
 
+	// kicks off the snapshot cleaning goroutine
+	go cleanSnapshots(cpClient)
+
 	// starts all of the listeners
 	zzk.Start(shutdown, conn, serviceListener, hostRegistry, snapshotListener)
 }
 
+func cleanSnapshots(cpClient dao.ControlPlane) {
```
Comment (smousa) [inline scheduler/leader.go]: need a shutdown channel

File: scheduler/leader.go [hunk 4]
```diff
@@ -57,13 +62,81 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, dao dao.Cont
...
+			// Get a list of services.
+			var services []service.Service
+			var serviceRequest dao.ServiceRequest
+			if err := cpClient.GetServices(serviceRequest, &services); err != nil {
+				glog.Warningf("Failed GetServices")
+				break
+			}
+
+			// Find the tenants.
+			var tenants []service.Service
```
Comment (smousa) [inline scheduler/leader.go]: Open a ticket to expose a function to return all the tenant services

**Comment by smousa** [comment]
If you set the TTL to 0, it should not clean snapshots

File: cli/cmd/cmd.go [hunk 5]
```diff
@@ -200,6 +200,7 @@ func New(driver api.API) *ServicedCli {
 		cli.IntFlag{"debug-port", configInt("DEBUG_PORT", 6006), "Port on which to listen for profiler connections"},
 		cli.IntFlag{"max-rpc-clients", configInt("MAX_RPC_CLIENTS", 3), "max number of rpc clients to an endpoint"},
 		cli.IntFlag{"rpc-dial-timeout", configInt("RPC_DIAL_TIMEOUT", 30), "timeout for creating rpc connections"},
+		cli.IntFlag{"snapshot-ttl", configInt("SNAPSHOT_TTL", 12), "snapshot TTL, 0 to disable"},
```
Comment (smousa) [inline cli/cmd/cmd.go]: need to specify unit

File: scheduler/leader.go [hunk 6]
```diff
@@ -57,13 +60,83 @@ func Lead(shutdown <-chan interface{}, conn coordclient.Connection, dao dao.Cont
...
 	serviceListener := zkservice.NewServiceListener(&leader)
 
+	// kicks off the snapshot cleaning goroutine
+	go cleanSnapshots(cpClient, snapshotTTL, shutdown)
+
 	// starts all of the listeners
 	zzk.Start(shutdown, conn, serviceListener, hostRegistry, snapshotListener)
 }
 
+func cleanSnapshots(cpClient dao.ControlPlane, snapshotTTL int, shutdown <-chan interface{}) {
```
Comment (smousa) [inline scheduler/leader.go]: suggestion: maybe pass in time.Duration?

**Pull request merged**